### PR TITLE
Compute functional enrichment revamp

### PR DESCRIPTION
**Pull request by meren**
This PR divides the following program:

* anvi-compute-functional-enrichment

Into three distinct programs:

* anvi-compute-functional-enrichment-across-genomes
* anvi-compute-functional-enrichment-in-pan
* anvi-compute-metabolic-enrichment

This was necessary to avoid some significant technical and design challenges to keep these quite distinct functionality served through a single program.

### CODE REVIEW

**Comment by meren** [comment]
@ivagljiva, with these changes, I believe the functions `functional_occurrence_stats` and `functional_enrichment_stats` in class `anvio/genomedescriptions:GenomeDescriptions` are no longer necessary as that code (and functions that are only called from within that code) is outsourced to `AggregateFunctions`. Am I correct?

**Comment by meren** [comment]
### TODO

- [ ] Update anvio/docs
- [ ] Update the pangenomics tutorial to use the fancy enrichment program

**Comment by ivagljiva** [comment]
@meren , yes you are correct :)

**Comment by meren** [comment]
I think I am done here. @ivagljiva, can you please take a look and let me know if you're satisfied with these changes?

File: bin/anvi-compute-functional-enrichment [hunk 1]
```diff
@@ -1,143 +1,31 @@
...
-__provides__ = ['functional-enrichment-txt',]
-__description__ = ("This is a driver program for `anvi-script-enrichment-stats`, a script that computes enrichment scores "
-                   "and group associations for annotated entities (ie, functions, KEGG Modules) across groups of genomes or "
-                   "samples.")
+def main():
+    run = Run()
+    run.warning("This program is broken into multiple programs. Our VERY FAVORITE one is "
+                "the metabolic enrichment one, but you feel free to choose the one from "
+                "the list of minions below based on what you wish to do:",
+                header="THIS PROGRAM IS NO MORE", lc="yellow")
```
Comment (ivagljiva) [inline bin/anvi-compute-functional-enrichment:18]: 😂 
anvi'o should really not be picking favorites from among her children, but I cannot say I disagree with her choice here :p

**Review by ivagljiva — APPROVED** [review]
@meren, I tested it and it is working _beautifully_. I particularly liked how genomes enrichment can now accept so many different types of input. thank you so much for doing this!

**Comment by meren** [comment]
Thank you very much for going through this and your fixes!

**Pull request merged**